### PR TITLE
Translation for Hooks : useReducer

### LINKED ISSUE #407: `useReducer`
Page: https://id.react.dev/reference/react/useReducer

Assignee: @rivldy

### PR DESCRIPTION
**Pull request by rivldy**
Hello, I just finished the translation. This is my first time translating, sorry if there are any word mistakes.

Page : https://id.react.dev/reference/react/useReducer
Closes #407

### CODE REVIEW

File: src/content/reference/react/useReducer.md [hunk 1]
```diff
@@ -20,7 +20,7 @@ const [state, dispatch] = useReducer(reducer, initialArg, init?)
 
 ### `useReducer(reducer, initialArg, init?)` {/*usereducer*/}
 
-Call `useReducer` at the top level of your component to manage its state with a [reducer.](/learn/extracting-state-logic-into-a-reducer)
+Panggil `useReducer` di atas komponen anda untuk mengelola statenya dengan [reducer.](/learn/extracting-state-logic-into-a-reducer)
```
Comment (r17x) [inline src/content/reference/react/useReducer.md:23]:
```suggestion
Panggil `useReducer` di tingkat atas komponen anda untuk mengelola statenya dengan [reducer.](/learn/extracting-state-logic-into-a-reducer)
```

File: src/content/reference/react/useReducer.md [hunk 2]
```diff
@@ -34,31 +34,31 @@ function MyComponent() {
...
-[See more examples below.](#usage)
+[Lihat contoh lainnya di bawah.](#usage)
 
-#### Parameters {/*parameters*/}
+#### Parameter {/*parameters*/}
 
-* `reducer`: The reducer function that specifies how the state gets updated. It must be pure, should take the state and action as arguments, and should return the next state. State and action can be of any types. 
-* `initialArg`: The value from which the initial state is calculated. It can be a value of any type. How the initial state is calculated from it depends on the next `init` argument.
-* **optional** `init`: The initializer function that should return the initial state. If it's not specified, the initial state is set to `initialArg`. Otherwise, the initial state is set to the result of calling `init(initialArg)`.
+* `reducer`: Fungsi reducer yang menentukan bagaimana state diperbarui. Itu harus murni, harus mengambil state dan action sebagai argumen, dan harus mengembalikan state berikutnya. State dan action bisa dari tipe apa saja.
```
Comment (r17x) [inline src/content/reference/react/useReducer.md:41]:
```suggestion
* `reducer`: Fungsi reducer yang menentukan bagaimana *state* diperbarui. Itu harus murni, harus mengambil *state* dan *action* sebagai argumen, dan harus mengembalikan *state* berikutnya. *State* dan *action* bisa dari tipe apa saja.
```

File: src/content/reference/react/useReducer.md [hunk 3]
```diff
@@ -34,31 +34,31 @@ function MyComponent() {
...
+* `initialArg`: Nilai dari mana initial state dihitung. Bisa menjadi nilai dari tipe apapun. Bagaimana initial state dihitung darinya bergantung pada argumen `init` berikutnya.
```
Comment (r17x) [inline src/content/reference/react/useReducer.md:42]: Seperti komentar sebelumnya (italic untuk istilah yang bukan bahasa indonesia).

**Review by r17x — DISMISSED** [review]

**Comment by mhaidarhanif** [comment]
I'll review again later.

**Review by r17x — APPROVED** [review]

**Review by resir014 — APPROVED** [review]

**Pull request merged**